sofana/R3BSofFrsAnalysis.cxx

```diff
@@ -302,12 +304,23 @@ void R3BSofFrsAnalysis::Exec(Option_t* option)
             i_s8cave = i;
         }
         if (isnan(tof) || tof < 0)
+        {
             continue;
-        beta[i] = fPathLength[i] / (tof + fTofOffset[i]); // ToFCalib
+        }
+        else
+        {
+            // beta[i] = fPathLength[i] / (tof + fTofOffset[i]); // ToFCalib
+            beta.push_back(fPathLength[i] / (tof + fTofOffset[i]));
+        }
     }
     // Velocity correlation conditions
-    if (fBetaCorr && beta[i_s2cave] > 0. && beta[i_s8cave] > 0. && TMath::Abs(beta[i_s2cave] - beta[i_s8cave] < 0.01))
-        beta[fNbTof] = beta[i_s2cave];
+    // Note that TMath::Abs(beta.at(i_s2cave) - beta.at(i_s8cave) < 0.01) should be TMath::Abs(beta.at(i_s2cave) -
+    // beta.at(i_s8cave)) < 0.01
+    if (fBetaCorr && beta.at(i_s2cave) > 0. && beta.at(i_s8cave) > 0. &&
+        TMath::Abs(beta.at(i_s2cave) - beta.at(i_s8cave) < 0.01))
+    {
+        beta.at(fNbTof) = beta.at(i_s2cave);
+    }
     //
     // Under modification
     int index_tof[4] = { fNbTof, i_s2cave, i_s8cave, i_s2s8 };
```

Comment:
Sorry for the poor way of coding but can we do this?
`int index_tof[4] = {beta.size()-1, i_s2cave, i_s8cave, i_s2s8 };`

Reply:
Because the line 306-314, the number of elements in the array is not always equal to fNbTof+1, it's necessary to replace fNbTof with beta.size()-1. Thank you.